### ISSUE TITLE
function Executor::blockSize() and Executor::gridSize() Logical error

### ISSUE THREAD

**Issue by thzyyx** [omnisci/mapd-core#294]
When obtaining the "blockSize" and "girdSize" of the graphics card information, different graphics card will only return the information of deviceID = 0

**Comment by randyzwitch** [comment]
Hi @thzyyx - 

How are you obtaining this information? Are you using pymapd, a different Thrift client, calling a method in the actual C++ code?

Thanks,
Randy

**Comment by thzyyx** [comment]
I found this problem while looking at the 4.2.0 source code.
unsigned Executor::blockSize() const {
  CHECK(catalog_);
  CHECK(catalog_->get_dataMgr().cudaMgr_);
  const auto& dev_props = catalog_->get_dataMgr().cudaMgr_->deviceProperties;
  return block_size_x_ ? block_size_x_ : dev_props.front().maxThreadsPerBlock;
}
"dev_props. Front ()" is the first element. If I have eight gpus, "dev_props" will have eight elements. Why different gpus always return the first element?

**Comment by asuhan** [comment]
@thzyyx You're right. However, we've decided against dealing with different GPUs on the same system. There'd be issues splitting the work effectively between GPUs with potentially very different performance. Also, it would prevent using the same native code back-end for all of them -- it's a lot of work for an uncommon scenario we don't really want to support.
